### PR TITLE
osutil/mkfs: Expose option for --lib flag in fakeroot call

### PR DESCRIPTION
**Pull request by GlenPickle**
ubuntu-image is a classic confined snap, and therefore has access to system files such as /usr/lib/. This has been causing problems with the existing fakeroot call, as the fakeroot binary was being called from within the snap, but loading libfakeroot from the host system.

While this is caused by fakeroot hard coding path values, fakeroot does provide a --lib flag to allow the user to specify another location to look for libfakeroot. The change in this PR uses an environment variable to look for the location of libfakeroot. That environment variable will be set in ubuntu-image, and then the correct libfakeroot can be loaded when ubuntu-image calls osutil/mkfs

I used the approach of an environment variable as it seemed like the minimal code change that would get the job done. If the snapd team would prefer I bubble up options that could be passed to mkfs.Ext4 instead, I would be happy to change strategies.

### CODE REVIEW

**Review by mvo5 — APPROVED** [review]
Thanks for digging into this! It looks like a reasonable workaround to me.

File: osutil/mkfs/mkfs.go
Comment (bboozzoo) [inline osutil/mkfs/mkfs.go:109]: Hm wonder if maybe it's more flexible to add a `FAKEROOT` env variable that holds the fakeroot invocation, eg.
```
FAKEROOT='fakeroot --lib=/foo/bar/baz' snap prepare-image ...
```
We have the code to parse this like a shell would to (see strutil/shlex package).

File: osutil/mkfs/mkfs.go
Comment (bboozzoo) [inline osutil/mkfs/mkfs.go:113]: Shouldn't we pass -- to clearly delineate fakeroot args from mkfs?
```suggestion
			mkfsArgs = append([]string{"--lib", fakerootLib, "--"}, mkfsArgs...)
```

File: osutil/mkfs/mkfs.go
Comment (sil2100) [reply osutil/mkfs/mkfs.go:109]: I of course leave this up to you, the snapd maintainers, to decide which approach is best, but I personally feel that just passing the library path is cleaner - although less flexible indeed. Do we expect to use other fakeroot options in the nearest time?

File: osutil/mkfs/mkfs.go
Comment (sil2100) [reply osutil/mkfs/mkfs.go:113]: This might indeed make things 'safer' - or at least cleaner. So +1

**Review by sil2100 — APPROVED** [review]
This change looks good to me as well! bboozzoo's recommended change to the fakeroot invocation command (with `--`) seems like a nice change to add, but not strictly required IMO.

**Comment by bboozzoo** [comment]
@mvo5 can you land it?

**Pull request merged**
